### PR TITLE
Wallet Core Integration Guide

### PR DESCRIPTION
**Pull request by catenocrypt**
Note that gitbook can be tried out locally:
```
git pull
git checkout ar/wallet-core-guide
npm install gitbook-cli -g
gitbook serve
```

### CODE REVIEW

File: SUMMARY.md [hunk 1]
```diff
@@ -2,13 +2,18 @@
...
-  * [Server-side](wallet-core/server-side.md)
+  * Using
+    * [Building](wallet-core/building.md)
+    * [WalletConsole utility](wallet-core/walletconsole.md)
+    * [Releasing](wallet-core/releasing.md)
+  * [Integration Guide](wallet-core/integration-guide.md)
+    * [Wallet Operations](wallet-core/wallet-operations.md)
+    * [iOS Wallet](wallet-code/ios-guide.md)
+    * [Server-Side](wallet-core/server-side.md)
+  * Extending
```
Comment (vikmeup) [inline SUMMARY.md]: what is extending?

File: wallet-core/ios-guide.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Sample iOS Wallet -- Step-by-step Guide
+
```
Comment (vikmeup) [inline wallet-core/ios-guide.md]: can we add this file later?

File: wallet-core/wallet-operations.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+The random generation employs secure random generation, as available on the device.
+
+```swift
+let wallet = HDWallet(strength: 128, passphrase: "")
+```
+
+Input parameter | Description
+---|---
+*strength* | The strength of the secret seed.  Higher seed means more information content, longer recovery phrase.  Default value is **128**, but 256 is also possible.
+*passphrase* | Optional passphrase, used to encrypt the seed.  If specified, the wallet can be imported and opened only with the passphrase (Not to be confused with recovery phrase).
```
Comment (vikmeup) [inline wallet-core/wallet-operations.md]: I don't think it's used for encrypted - it's more like offset?

File: wallet-core/wallet-operations.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+let wallet = HDWallet(mnemonic: "ripple scissors kick mammal hire column oak again sun offer wealth tomorrow wagon turn fatal", passphrase: "")
+```
+
+Input parameter | Description
+---|---
+*mnemonic* | a.k.a. *recovery phrase*.  The string of several words that was used to create the wallet.
+*passphrase* | Optional passphrase, used to encrypt the seed.
+
+---------------------------
+TODO: fill operations guide
```
Comment (vikmeup) [inline wallet-core/wallet-operations.md]: can we remove todo?

File: SUMMARY.md (hunk 1 above)
Comment (catenocrypt) [reply SUMMARY.md]: Extending with new functionality, e.g. with new blockchain.  Some devs are only using WC (wallet devs), some extend it (e.g. add new blockchain).

File: wallet-core/ios-guide.md (hunk 2 above)
Comment (catenocrypt) [reply wallet-core/ios-guide.md]: This is early WIP, I created PR so others can see it.

File: wallet-core/wallet-operations.md (hunk 3 above)
Comment (catenocrypt) [reply wallet-core/wallet-operations.md]: Good point, I need to check into this...  I think 'scramble' is a good term to use here :)

File: wallet-core/ios-guide.md (hunk 2 above)
Comment (vikmeup) [reply wallet-core/ios-guide.md]: for WIP use draft mode for PR's

File: SUMMARY.md [hunk 5]
```diff
@@ -2,13 +2,18 @@
...
 * [Get Started](README.md)
 * [Wallet Core](wallet-core/wallet-core.md)
-  * [Building](wallet-core/building.md)
-  * [Contributing](wallet-core/contributing.md)
-  * [Adding Support for a New Blockchain](wallet-core/newblockchain.md)
-  * [RPC / API Requirements](wallet-core/rpc-requirements.md)
-  * [Releasing](wallet-core/releasing.md)
-  * [WalletConsole utility](wallet-core/walletconsole.md)
-  * [Server-side](wallet-core/server-side.md)
+  * Using
```
Comment (hewigovens) [inline SUMMARY.md]: maybe Developing?

File: wallet-core/wallet-operations.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
+# Wallet Operations Guide
+
+We present here an overview of the basic wallet operations.  More detailed, language-specific samples are provided in step-by-step [guides](wallet-core/integration-guide.md).
+
+The covered basic operations are:
+
+* Wallet management
+  * Creating a new multi-coin wallet
```
Comment (hewigovens) [inline wallet-core/wallet-operations.md]:
```suggestion
  * Creating a new multi-coin wallet (HD wallet)
```

File: wallet-core/wallet-operations.md [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+  * Generating an address using a custom derivation path (expert)
+* Transaction signing (sending)
+  * Coin-specific transaction signing
+  * Coin-independent method ('AnySigner')
+
+For the examples we use *Bitcoin*, *Ethereum* and *Binance Coin* as sample coins/blockchains.
+
+Note: Wallet Core does not cover communication with blockchain networks (nodes): 
+address derivation is covered, but address balance retrieval not; 
+transaction signing is convered, but broadcasting transactions to the network not.
```
Comment (hewigovens) [inline wallet-core/wallet-operations.md]:
```suggestion
transaction signing is covered, but broadcasting transactions to the network not.
```

File: wallet-core/wallet-operations.md [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+
+If the wallet was created with a passphrase, it is also required.
+
+```swift
+let wallet = HDWallet(mnemonic: "ripple scissors kick mammal hire column oak again sun offer wealth tomorrow wagon turn fatal", passphrase: "")
+```
+
+Input parameter | Description
+---|---
+*mnemonic* | a.k.a. *recovery phrase*.  The string of several words that was used to create the wallet.
```
Comment (hewigovens) [inline wallet-core/wallet-operations.md]: mention that we're BIP39 compatible

File: wallet-core/wallet-operations.md [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+*passphrase* | Optional passphrase, used to encrypt the seed.
+
+## Account Address Derivation
+
+Each coin needs a different account, with matching address.  Addresses are derived from the multi-coin wallet.
```
Comment (hewigovens) [inline wallet-core/wallet-operations.md]: mention that we're bip44/bip84 compatible

File: wallet-core/wallet-operations.md [hunk 10]
```diff
@@ -90,10 +98,83 @@ This can be done in two steps: first a derived private key is obtained, then an
...
+
+A generic, coin-independent signer also exists (*AnySigner*), but its usage is recommended only in browser-based applications.
+
+### Ethereum Transaction Signing
+
+A simple Ethereum send transaction needs the following fields:
+
+Field | Sample value | Description
+---|---|---
+chainID | 1 | Network selector, use 1 for mainnet
```
Comment (kolya182) [inline wallet-core/wallet-operations.md]: You can mention source for other chains https://chainid.network/

Comment (catenocrypt) [reply wallet-core/wallet-operations.md]: 👍

File: wallet-core/wallet-operations.md [hunk 11]
```diff
@@ -0,0 +1,182 @@
...
+gasPrice | 3600000000 | The price to determine the amount of ether the transaction will cost
+gasLimit | 21000 | The maximum gas that is allowed to be spent to process the transaction
+to | &lt;address&gt; | The account the transaction is sent to, if empty, the transaction will create a contract
+value | 100000000 | The amount of ether to send
+data | | Could be an arbitrary message or function call to a contract or code to create a contract
+
+Code example to fill in the signer input parameters:
+
+```swift
+let signerInput = TW_Ethereum_Proto_SigningInput.with {
```
Comment (hewigovens) [inline wallet-core/wallet-operations.md]:
```suggestion
let signerInput = EthereumSigningInput.with {
```

File: wallet-core/wallet-operations.md [hunk 12]
```diff
@@ -0,0 +1,182 @@
...
+Field | Sample value | Description
+---|---|---
+chainID | 1 | Network selector, use 1 for mainnet (see https://chainid.network for more)
+nonce | 1 | The count of the number of outgoing transactions, starting with 0
+gasPrice | 3600000000 | The price to determine the amount of ether the transaction will cost
+gasLimit | 21000 | The maximum gas that is allowed to be spent to process the transaction
+to | &lt;address&gt; | The account the transaction is sent to, if empty, the transaction will create a contract
+value | 100000000 | The amount of ether to send
+data | | Could be an arbitrary message or function call to a contract or code to create a contract
+
```
Comment (hewigovens) [inline wallet-core/wallet-operations.md]: chain id, nonce, gas can get from RPC, https://github.com/ethereum/wiki/wiki/JSON-RPC

File: wallet-core/wallet-operations.md [hunk 13]
```diff
@@ -0,0 +1,182 @@
...
+* https://kauri.io/article/7e79b6932f8a41a4bcbbd194fd2fcc3a/v2/ethereum-101-part-4-accounts-transactions-and-messages
+* https://github.com/ethereumbook/ethereumbook/blob/develop/06transactions.asciidoc
+
+### Coin-Independent Signing ('AnySigner')
+
+Wallet Core also has a generic, coin-independent signer.  The transaction fields are provided in a JSON string, with the same content.  Since this is not strongly typed, its usage is not recommended.  We reproduce here one example (with Ethereum) nonetheless.
+
+```swift
+let transaction =  "{\"chainId\":\"AQ==\",\"gasPrice\":\"1pOkAA==\",\"gasLimit\":\"Ugg=\",\"toAddress\":\"" + dummyReceiverAddress + "\",\"amount\":\"A0i8paFgAA==\"}"
+let anySignerInput = TW_Any_Proto_SigningInput.with {
```
Comment (hewigovens) [inline wallet-core/wallet-operations.md]:
```suggestion
let anySignerInput = AnySigningInput.with {
```

File: wallet-core/wallet-operations.md [hunk 14]
```diff
@@ -0,0 +1,182 @@
...
+It is also possible to derive addresses using custom derivation paths.
+This can be done in two steps: first a derived private key is obtained, then an address from it.
+
+> **Warning**: use this only if you are well aware of the semantics of the derivation path used!
+
+> **Security Warning**: if secrets such as private keys are handled by the wallet, even if for a short time, handle with care!
+> Avoid any risk of leakage of secrets!
+
+```swift
+let key = wallet.getKey(derivationPath: "m/44\'/60\'/1\'/0/0")   // m/44'/60'/1'/0/0
```
Comment (hewigovens) [inline wallet-core/wallet-operations.md]:
```suggestion
let key = wallet.getKey(derivationPath: "m/44'/60'/1'/0/0")   // m/44'/60'/1'/0/0
```

Comment (catenocrypt) [reply wallet-core/wallet-operations.md]: Oddly this did not work for me, it gave an exception that components must be separated by '/'.  I tried escaping the 's, and then it worked.

**Pull request merged**